packages/react-db/src/useLiveQuery.ts

```diff
@@ -280,6 +290,26 @@ export function useLiveQuery<
   isEnabled: true // Always true for pre-created live query collections
 }
 
+// Overload 8: Accept pre-created live query collection with single: true
+export function useLiveQuery<
+  TResult extends object,
+  TKey extends string | number,
+  TUtils extends Record<string, any>,
+>(
+  liveQueryCollection: Collection<TResult, TKey, TUtils> & { single: true }
```

Comment:
We should probably have helper types instead of writing `& { single: true }` or `& { single: never }` in a bunch of places. That will greatly simplify maintainability in case we change something to this later.